controllers/bgImgController.js

```diff
@@ -0,0 +1,125 @@
+import { httpStatus } from "../utils/httpStatusCode.js";
+import { RestaurantBgImg } from "../models/bgImgModel.js";
+import mongoose from "mongoose";
+
+const getBgImgs = async (req, res) => {
+    try {
+        const bgImgs = await RestaurantBgImg.find(req.query);
+
+        if (bgImgs.length === 0) {
+            return res.status(httpStatus.NO_CONTENT).json({
+                message: "No background image found",
+            });
+        }
+
+        return res.status(httpStatus.OK).json({
+            success: true,
+            data: bgImgs,
+        });
+    } catch (err) {
+        return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+            success: false,
+            error: "Server Error",
+        });
+    }
+};
+
+const addBgImg = async (req, res) => {
+    const bgImg = new RestaurantBgImg({
+        _id: new mongoose.Types.ObjectId(),
+        backgroundImage: req.file.filename,
+    });
+    try {
+        const createdBgImg = await RestaurantBgImg.create(bgImg);
+
+        return res.status(httpStatus.CREATED).json({
+            success: true,
+            data: createdBgImg,
+        });
+    } catch (err) {
+        return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+            success: false,
+            error: "Server Error",
+        });
+    }
+};
+
+const getBgImgById = async (req, res) => {
+    try {
+        const bgImg = await RestaurantBgImg.findById(req.params.id);
+        if (!bgImg) {
```

Comment:
does it make sens to return no content when there is no image in db with this id ? 
This rout probably is something like this /img/4 not imag?id=1. In this case i would return 404